### PR TITLE
Feature: Generators

### PR DESCRIPTION
**Pull request by joernweissenborn**
This PR salvages the generators from the project PR.

This is kind of in conflict with the work of @jsnel but we should use this PR to marry everything.

My approach of generators is more "dump", as the generating functions should return model dicts instead of models, I'am not sure why, but I think this will be useful. 

Also this generators no't have any state, they are boilerplate replacement and I think all what is needed.

The usage will become more  clear in a follow up PR where I add projects. This will also enable all the features the current Generator achieves with state fullness. 

More details to follow
### Change summary

- Created `glotaran.project.generators` package

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)
- [ ] 👌 Closes issue (mandatory for ✨ feature and 🩹 bug fix PR's)
- [x] 🧪 Adds new tests for the feature (mandatory for ✨ feature and 🩹 bug fix PR's)
- [x] 📚 Adds documentation of the feature

### Closes issues

closes #XXXX

### CODE REVIEW

File: glotaran/deprecation/modules/test/test_project_result.py
Comment (s-weigand) [inline glotaran/deprecation/modules/test/test_project_result.py:19]:
```suggestion
def test_result_get_dataset_method(dummy_result: Result):
```
Originally my bad naming but let's fix it anyway so sonarcloud won't complain anymore ^^

File: glotaran/deprecation/modules/test/test_project_result.py
Comment (s-weigand) [inline glotaran/deprecation/modules/test/test_project_result.py:29]:
```suggestion
def test_result_get_dataset_method_error(dummy_result: Result):
```
Originally my bad naming but let's fix it anyway so sonarcloud won't complain anymore ^^

File: glotaran/testing/parallel_spectral_decay.py
Comment (s-weigand) [inline glotaran/testing/parallel_spectral_decay.py:18]: `SIMULATION_PARAMETER_YML` would be a more fitting name since the actually wanted parameters are `PARAMETER_YML`.
That said we could also reuse the `PARAMETER_YML` To reduce duplication and improve readability.

File: glotaran/testing/sequential_spectral_decay.py
Comment (s-weigand) [inline glotaran/testing/sequential_spectral_decay.py:18]: `SIMULATION_PARAMETER_YML` would be a more fitting name since the actually wanted parameters are `PARAMETER_YML`.
That said we could also reuse the `PARAMETER_YML` To reduce duplication and improve readability.

File: glotaran/testing/parallel_spectral_decay.py
Comment (s-weigand) [inline glotaran/testing/parallel_spectral_decay.py:57]: Rather create a constant `SIMULATION_COORDINATES` and share import it to reduce code duplication.

```python 
SIMULATION_COORDINATES = {"time": np.arange(-1, 20, 0.01), "spectral": np.arange(600, 700, 1.4)}
```

**Review by s-weigand — CHANGES_REQUESTED** [review]
Mostly nitpicky things

Please move
- `glotaran/testing/parallel_spectral_decay.py` -> `glotaran/testing/simulated_data/parallel_spectral_decay.py`
- `glotaran/testing/sequential_spectral_decay.py` -> `glotaran/testing/simulated_data/sequential_spectral_decay.py`

That way we can add a file `glotaran/testing/simulated_data/common.py` and share common values (e.g. `SIMULATION_COORDINATES`, `PARAMETER_YML`), which will reduce code duplication.

As for disabling the `quick-start` integration test (ca82e91), we should revert that commit and properly deprecate `glotaran.examples` that is what `glotaran.deprecation.modules` is for.

**Comment by s-weigand** [comment]
@joernweissenborn Don't forget to add the changes to the changelog. (@jsnel we still need to add the changes from #860)

File: glotaran/project/generators/generator.py
Comment (joernweissenborn) [reply glotaran/project/generators/generator.py]: changed to ue the builtin yml model.

File: glotaran/testing/parallel_spectral_decay.py
Comment (joernweissenborn) [reply glotaran/testing/parallel_spectral_decay.py]: It is there as yaml for user inspection and copy.

File: glotaran/project/generators/generator.py
Comment (joernweissenborn) [reply glotaran/project/generators/generator.py:189]: This will get extended to generate more then decay models, thus the arguments are arbitrary and generator specific.

**Comment by s-weigand** [comment]
I made the generators keyword only and added a `TypedDict` for the arguments.
@joernweissenborn  please check that you are ok with this change.

**Review by s-weigand — DISMISSED** [review]
From my side it is fine now 🚀

**Review by s-weigand — APPROVED** [review]
Now that `glotaran.examples` is properly deprecated and all integration tests run again it should be good to merge.

File: glotaran/builtin/io/yml/test/test_save_scheme.py
Comment (jsnel) [inline glotaran/builtin/io/yml/test/test_save_scheme.py:15]: This should be `PARAMETERS`.

File: glotaran/deprecation/deprecation_utils.py
Comment (jsnel) [inline glotaran/deprecation/deprecation_utils.py:614]: This function could do with some cleaning up in terms of documentation, @s-weigand.

File: glotaran/deprecation/modules/test/test_changed_imports.py
Comment (jsnel) [inline glotaran/deprecation/modules/test/test_changed_imports.py:136]: -> "wanted_parameters", "parameters"

File: glotaran/deprecation/modules/test/test_changed_imports.py
Comment (jsnel) [inline glotaran/deprecation/modules/test/test_changed_imports.py:139]: addributes -> attributes

File: glotaran/testing/simulated_data/shared_decay.py
Comment (jsnel) [inline glotaran/testing/simulated_data/shared_decay.py:42]: `PARAMETER` -> `PARAMETERS`

https://simple.wiktionary.org/wiki/parameter

**Review by jsnel — COMMENTED** [review]
Looks good to me, just some minor cosmetic changes requested (not all of them captured by the comments), but I will pick that up myself and then give it the green light.

**Review by jsnel — APPROVED** [review]
Can be merged in now! 🎉 

Note to @glotaran/maintainers: we should revisit #815 after merging in this PR.

**Comment by jsnel** [comment]
Update, last messages was written under the assumption I merged in some changes, which I didn't. Migrated those changes to a separate follow up PR:
https://github.com/glotaran/pyglotaran/pull/965

**Pull request merged**
